pkg/psmdb/mongos.go

```diff
@@ -29,7 +42,7 @@ func MongosDeployment(cr *api.PerconaServerMongoDB) *appsv1.Deployment {
 	}
 }
 
-func MongosDeploymentSpec(cr *api.PerconaServerMongoDB, operatorPod corev1.Pod, log logr.Logger, customConf CustomConfig, cfgInstances []string) (appsv1.DeploymentSpec, error) {
```

Comment:
I think we should replace Deployment only when CR is updated from old version to new version

Reply:
please don't forget to remove deployment during update